Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py

```diff
@@ -86,13 +120,138 @@ def http_request(self, feed_type) -> list:
+            return None
+
+        autofocus_indicator_type = artifact.get('indicator_type', None)
+        indicator_type = af_indicator_type_to_demisto.get(autofocus_indicator_type)
+        if not indicator_type:
+            return None
+
+        raw_json_data.update(
+            {
+                'value': indicator_value,
+                'type': indicator_type,
+                'confidence': artifact.get('confidence', ''),
+                'autofocus_malware': artifact.get('m', 0),
+                'autofocus_benign': artifact.get('b', 0),
+                'autofocus_grayware': artifact.get('g', 0)
+            }
+        )
+
+        if indicator_type == FeedIndicatorType.IP and ':' in indicator_value:
+            indicator_value, port = indicator_value.split(':', 1)
+            raw_json_data['autofocus_port'] = port
+
+        return {
+            'value': raw_json_data['value'],
+            'type': raw_json_data['type'],
+            'rawJSON': raw_json_data
+        }
+
+    @staticmethod
+    def create_indicators_from_single_sample_response(single_sample):
+        single_sample_data = single_sample.get('_source', {})
+        if not single_sample_data:
+            return []
+
+        artifacts = single_sample_data.get('artifact', [])
+
+        raw_json_data = {
+            'autofocus_id': single_sample.get('_id'),
+            'autofocus_region': single_sample_data.get('region', []),
+            'autofocus_malware': single_sample_data.get('malware', 0),
+            'autofocus_tags': single_sample_data.get('tag', []),
+            'autofocus_tags_groups': single_sample_data.get('tag_groups', []),
+            'autofocus_num_matching_artifacts': len(artifacts),
+        }
+
+        update_date = single_sample_data.get('update_date', None)
+        if update_date is not None:
+            update_date = datetime.strptime(update_date, '%Y-%m-%dT%H:%M:%S')
+            raw_json_data['autofocus_update_date'] = datetime_to_epoch(update_date)
+
+        create_date = single_sample_data.get('update_date', None)
+        if create_date is not None:
+            create_date = datetime.strptime(create_date, '%Y-%m-%dT%H:%M:%S')
+            raw_json_data['autofocus_create_date'] = datetime_to_epoch(create_date)
+
+        sha256 = single_sample_data.get('sha256', '...')
+        sha1 = single_sample_data.get('sha1', None)
+        md5 = single_sample_data.get('md5')
+
+        if '...' in sha256:
+            return []
+
+        indicators = [
+            {
+                'value': file_indicator,
+                'type': FeedIndicatorType.File,
+                'rawJSON': raw_json_data
+            }
+            for file_indicator in [sha256, sha1, md5] if file_indicator
+        ]
+
+        for artifact in artifacts:
+            indicator_from_artifact = Client.create_indicator_from_artifact(raw_json_data, artifact)
+            if indicator_from_artifact:
+                indicators.append(indicator_from_artifact)
```

Comment:
didn't we say each artifact is a list? so it should return a list of indicators, doesn't it?

Reply:
Each artifact is an indicator.
The source is the big file which holds all the artifacts.